jupyterhub/templates/proxy/deployment.yaml

```diff
@@ -65,13 +72,18 @@ spec:
             - --redirect-port=8000
             - --ssl-key=/etc/chp/tls/tls.key
             - --ssl-cert=/etc/chp/tls/tls.crt
+            {{- else if $manualHTTPSwithsecret }}
+            - --port=8443
+            - --redirect-port=8000
+            - --ssl-key=/etc/chp/tls/{{ .Values.proxy.https.secret.key }}
+            - --ssl-cert=/etc/chp/tls/{{ .Values.proxy.https.secret.crt }}
```

Comment:
This PR wanted to utilize a pre-existing secret containing a key/crt for TLS. This means that this PR should

1. Ensure that we can override the secret name.
2. Ensure we don't create a secret with TLS content any more.

This PR is now overriding the secret name, mounting it etc, but then letting the container utilize the key/crt through command line arguments anyhow. This means that the provided secret mounted is never utilized anyhow.

Note that the secret is made available on line 48-53, and it is mounted to the `chp` container on line 86.

---

If a secret is to be provided though, it must have the same keys that we have, which isn't obvious though. @konfiot, what is the structure of the pre-existing secret that you wanted to utilize?

Does it happen to be formatted exactly like this, having `tls.crt` and `tls.key` as keys for the secrets `data` dictionary?
```yaml
data:
  tls.crt: <base-64-encoded cert>
  tls.key: <base-64-encoded key>
```

Reply:
To summarize @konfiot, if you pass the crt / key file anyhow, you could utilize the `manual` `proxy.https.type`, and if you want to utilize a preexisting secret, the following lines should reference the files that the secret has mounted in `/etc/chp/tls`. You must know the files names though, and that depends on what you have named the keys within your pre-existing secrets `data` dictionary. Every key in the `data` dictionary will become a file.